prow/external-plugins/needs-rebase/plugin/plugin.go

```diff
@@ -68,27 +69,39 @@ The plugin reacts to commit changes on PRs in addition to periodically scanning
 
 // HandlePullRequestEvent handles a GitHub pull request event and adds or removes a
 // "needs-rebase" label based on whether the GitHub api considers the PR mergeable
-func HandlePullRequestEvent(log *logrus.Entry, ghc githubClient, pre *github.PullRequestEvent) error {
+func HandlePullRequestEvent(log *logrus.Entry, ghc githubClient, pre *github.PullRequestEvent, cache *Cache) error {
 	if pre.Action != github.PullRequestActionOpened && pre.Action != github.PullRequestActionSynchronize && pre.Action != github.PullRequestActionReopened {
 		return nil
 	}
-
-	return handle(log, ghc, &pre.PullRequest)
+	if cache.validTime > 0 && cache.Get(pre.PullRequest.ID) {
+		return nil
+	}
+	err := handle(log, ghc, &pre.PullRequest)
+	if cache.validTime > 0 && err == nil {
+		cache.Set(pre.PullRequest.ID)
+	}
+	return err
```

Comment:
I don't think we should be using the cache when handling `PullRequestEvent` - these are actions that directly affect mergeability, so IMO not checking mergeabiltiy would be a mistake. I think reading a value from cache only makes sense on handling comments (because comments do not affect mergeability, they just make us re-check)